File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.length() == 0) return true;
        int matchedChars = 0;
        for (int i = 0; i < t.size(); ++i) { 
            if (matchedChars < s.size() && s[matchedChars] == t[i]) matchedChars++;
        }
        return matchedChars == s.size();
    }
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if not s:
            return True
        sStack = list(s)

        for i in range(len(t)-1, -1, -1):
            if t[i] == sStack[-1]:
                sStack.pop()

            if not sStack:
                return True

        return False